pkg/operator/resourcesynccontroller/conditional_resourcesync_controller.go

```diff
@@ -0,0 +1,83 @@
+package resourcesynccontroller
+
+import (
+	"context"
+	"time"
+
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+// preconditionsFulfilled is a function that indicates whether all prerequisites
+// are met and the controller can Sync.
+type preconditionsFulfilled func() (bool, error)
+
+// ConditionalResourceSyncController is a controller that will copy source
+// configmaps and secrets to their destinations. It will also mirror deletions
+// by deleting destinations.
+// It will only do that if the given preconditions are fulfilled.
+type ConditionalResourceSyncController struct {
+	*ResourceSyncController
+
+	preconditionsFulfilledFn preconditionsFulfilled
+}
+
+var _ ResourceSyncer = &ConditionalResourceSyncController{}
+var _ factory.Controller = &ConditionalResourceSyncController{}
+
+// NewConditionalResourceSyncController creates NewConditionalResourceSyncController.
+func NewConditionalResourceSyncController(
```

Comment:
couldn't this also be by resource and not with another controller? How do both implementation compare?

Reply:
I think it would make sense to have by resource instead of defining one global condition for the whole controller. Going this route there wouldn't be much difference between both controllers except that one adds a condition per resource, so I think it would make sense to extend the original resource sync controller instead of adding a new one.

Reply:
I updated the PR to extend the ResourceSyncer interface and add the `SyncResourceConditionally` functions. PTAL @sttts @p0lyn0mial